.githooks/pre-commit

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# this shell script is stored as .githooks/pre-commit
+
+UID=$(id -u)
+
+echo 'Check entangled files are up to date'
+
+FILES=$(docker run --rm --user ${UID} -v ${PWD}:/data nlesc/pandoc-tangle:0.5.0 --preserve-tabs *.md 2>&1 > /dev/null | perl -ne 'print $1,"\n" if /^Writing \`(.*)\`./')
```

Comment:
This looks very cryptic but probably you couldn't find a workaround.

Reply:
Split it up and added comments